packages/document/builder-doc/docs/zh/guide/advanced/source-build.md

```diff
@@ -1,15 +1,79 @@
 # 源码构建模式
 
-源码构建模式允许开发者在 Monorepo 场景下使用依赖的子项目源码进行开发。这样可以在不启动子项目构建任务的情况下进行 HMR (Hot Module Replacement)。
+源码构建模式用于 monorepo 开发场景，它允许开发者直接引用 monorepo 中其他子项目的源码进行开发。
 
-## 开启特性
+## 使用场景
 
-你可以通过设置 [`experiments.sourceBuild`](/api/config-experiments.html#experimentssourcebuild) 为 `true` 来开启该功能。
+**monorepo 中项目互相引用主要有产物引用和源码引用两种方式**。我们以一个最简单的 monorepo 为例子，来介绍源码引用的使用场景。
 
-## 指定需要读取源码的子项目
+比如 monorepo 中包含了一个 app 应用和一个 lib 包：
 
-当需要读取某一个子项目源码的时候，需要确保子项目的 package.json 中包含 `source` 字段，并且该字段对应的文件路径为源码文件路径。
+```ts
+monorepo
+├── app
+└── lib
+    └── src
+        └── index.ts
+```
+
+其中，app 是基于 Modern.js Builder 构建的，app 依赖了 lib 中的一些方法：
+
+```json
+{
+  "name": "app",
+  "dependencies": {
+    "lib": "workspace:*"
+  }
+}
+```
+
+### 产物引用
+
+**产物引用指的是当前项目引用其他子项目构建后的产物。**
+
+比如上述例子中的 lib 是使用 TypeScript 编写的，通常情况下，我们需要提前构建 lib 的代码，生成 JavaScript 产物，这样 app 才可以正确引用它。当 lib 代码更新时，还需要重新执行一次构建（或者使用 `tsc --watch`），否则 app 无法引用到最新的代码。
+
+这种方式的优势在于：
+
+- 各个子项目的构建过程是完全独立的，可以拥有不同的构建配置。
+- 可以针对子项目进行构建缓存。
+
+劣势在于：
+
+- 本地开发时 HMR 的链路变长。
+- 当一个项目中包含多个 lib 包时，以上过程会显得十分繁琐。
 
+### 源码引用
+
+**源码引用指的是当前项目引用其他子项目的源码进行构建。**
+
+比如上述例子，当你开启了源码构建模式时，Modern.js Builder 会自动引用 lib 的 `src/index.ts` 源代码。这意味着，你不需要提前构建 lib 的代码，并且当 lib 的源代码更新时，也可以自动触发 app 的热更新。
+
+这种方式的优势在于：
+
+- 子项目不依赖构建工具，也不需要添加构建配置，子项目的代码会被当前项目的构建工具编译。
+- 不需要提前执行子项目的构建流程。
+- 本地开发时 HMR 更高效。
```

Comment:
Should we also mention the disadvantages, such as inconsistent behavior in different host applications and higher cost for code transpiling? If used for debugging component libraries, the problem will be even more complex. For example, the default behavior of CSS modules in app tools and module tools may differ, and even with custom configurations, complete alignment may be impossible.

Reply:
Nice catch, I will add some disadvantages here

Reply:
I have added some disadvantages, and I do not mention the `module-tools` because most of the behaviors are same, and we should align the behaviors in the future.